go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -362,6 +385,17 @@ func (te *TxEngine) shutdownLocked() {
 	log.Infof("TxEngine - making sure the goroutine has returned")
 	<-rollbackDone
 
+	// Shut down functions are idempotent.
+	// No need to check if 2pc is enabled.
+	// We stop the transaction watcher so late, because if the user isn't running
+	// with any shutdown grace period, we still want the watcher to run while we are waiting
+	// for resolving transactions.
+	log.Infof("TxEngine - stop transaction watcher")
+	te.stopTransactionWatcher()
+
+	// Mark the prepared pool closed.
+	log.Infof("TxEngine - closing the prepared pool")
+	te.preparedPool.Close()
 	log.Infof("TxEngine - closing the txPool")
 	te.txPool.Close()
```

Comment:
we should follow the reverse order of open during close.
txpool close > twoPC close > preparePool close

Reply:
I know there is zero impact of the order here. But, makes it simple to reason about.